pulp_ansible/app/tasks/collections.py

```diff
@@ -608,13 +663,60 @@ async def _find_all_collections(self):
 
         await asyncio.gather(*tasks)
 
+    async def _should_we_sync(self):
+        """Check last synced metadata time."""
+        root, api_version = await self._get_root_api(self.remote.url)
+        if api_version == 3:
+            downloader = self.remote.get_downloader(
+                url=root, silence_errors_for_response_status_codes={404}
+            )
+            try:
+                metadata = parse_metadata(await downloader.run())
+            except FileNotFoundError:
+                return False
+
+            try:
+                self.last_synced_metadata_time = parse_datetime(metadata["published"])
+            except KeyError:
+                return False
+
+            sources = set()
+            if self.collection_info:
+                sources = {r.source for r in self.collection_info if r.source}
+            sources.add(self.remote.url)
+            if len(sources) > 1:
+                return False
+
+            if self.last_synced_metadata_time == self.repository.last_synced_metadata_time:
+                return True
+
+        return False
+
     async def run(self):
         """
         Build and emit `DeclarativeContent` from the ansible metadata.
         """
+        if self.repository.remote and self.remote.pk == self.repository.remote.pk:
+            can_be_optimized = await self._should_we_sync()
```

Comment:
```suggestion
            should_we_sync = await self._should_we_sync()
```
This name aligns with the True/False value it's receiving.